### ISSUE TITLE
multi_scan doesn't collate results when it uses a single batch

### ISSUE THREAD

**Issue by lilyball** [lyndsey-ferguson/fastlane-plugin-test_center#254]
### New Issue Checklist

- [X] Updated `fastlane-plugin-test_center` to the latest version
- [X] I read the [README.md](https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/blob/master/README.md)
- [X] I reviewed the [example(s)](https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/blob/master/README.md) for the action(s) I am using
- [X] I have removed any sensitive data such as passwords, authentication tokens, or anything else I do not want to world to see

> If you love this fastlane plugin, consider sponsoring it or asking your company to sponsor it. I would really appreciate any
> gesture: https://github.com/sponsors/lyndsey-ferguson. 😍

### Issue Description
I'm invoking `multi_scan` with the following options:

```
try_count: 2, # retry failed tests x times
batch_count: 1, # divide tests into x batches
parallel_testrun_count: 4, # run on x simulators
```

However if I also pass `only_testing` with a single test, multi_scan doesn't behave as expected. In particular, despite the single test, it still creates 4 simulators (based on the fact that it deletes 4 when it's done), spins up a worker, runs a single batch on the worker, and then exits without attempting to collate that single batch. That leaves the output in an unexpected state; instead of `test_output/TargetUITests/report-files-go-here` I'm left with `test_output/TargetUITests/TargetUITests-batch-1/report-files-go-here`, and the collected log archive also has the `batch-1-try-0-` prefix.

The simple fix would be to just "collate" the single batch, moving the files upwards. Though really it would be better if multi_scan would realize it only has 1 batch to run and disable batching entirely. I'm not sure how retries interact with batching though. Also it shouldn't create more simulators than it actually needs; even if I were to do 3 batches, it shouldn't create 4 simulators.

**Comment by lyndsey-ferguson** [comment]
@lilyball can you modify your `Pluginfile` per my instructions below, run `bundle install`, and then run your fastlane again (with the `--verbose` flag)?

Pluginfile:
```
gem 'fastlane-plugin-test_center', :git => "https://github.com/lyndsey-ferguson/fastlane-plugin-test_center.git", :branch => "issue-254-wonky-batches"
```

If there are still problems, please let me know and attach the console output as a text file to this issue (makes it easier for me to review).  If it works, please let me know.

**Comment by lyndsey-ferguson** [comment]
I moved this back into progress as I've discovered an issue.

**Comment by lyndsey-ferguson** [comment]
Can you modify your `Pluginfile` per my instructions below, run `bundle install`, and then run your fastlane again (with the `--verbose` flag)?

Pluginfile:
```
gem 'fastlane-plugin-test_center', :git => "https://github.com/lyndsey-ferguson/fastlane-plugin-test_center.git", :branch => "issue-254-wonky-batches"
```

If there are still problems, please let me know and attach the console output as a text file to this issue (makes it easier for me to review).  If it works, please let me know.

**Comment by lyndsey-ferguson** [comment]
@lilyball I would appreciate it if you could test this: I'd like to confirm that this addresses your concerns. If you're too busy right now, I can merge this on Friday and we can open up another issue if there are remaining problems.

**Comment by lyndsey-ferguson** [comment]
Fixed in [v3.11.6](https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/releases/tag/v3.11.6)
